ports/esp32/machine_hw_spi.c

```diff
@@ -269,6 +269,16 @@ STATIC void machine_hw_spi_deinit(mp_obj_base_t *self_in) {
     }
 }
 
+STATIC mp_uint_t gcd(mp_uint_t x, mp_uint_t y) {
```

Comment:
can you please make this a loop rather than recursive, just in case the compiler cannot eliminate the tail recursion

Reply:
Sure. I was being lazy here and just grabbing the first/simplest GCD implementation I could find. I wasn't sure if I'd be instructed to actually put it somewhere else as a utility function.